src/tasks/handlers/baseHandler.ts

```diff
@@ -0,0 +1,185 @@
+import {
+  ICreateTaskBody,
+  IJobResponse,
+  ITaskResponse,
+  JobManagerClient,
+  OperationStatus,
+  TaskHandler as QueueClient,
+} from '@map-colonies/mc-priority-queue';
+import { Logger } from '@map-colonies/js-logger';
+import { BadRequestError, ConflictError } from '@map-colonies/error-types';
+import { IConfig, IJobDefinitionsConfig, JobAndTask, TaskTypesArray } from '../../common/interfaces';
+import { JOB_COMPLETED_MESSAGE } from '../../common/constants';
+import { taskParameterMapper } from '../../common/mappers';
+import { calculateTaskPercentage } from '../../utils/taskUtils';
+import { isInitialWorkflowCompleted } from './utils';
+
+export abstract class JobHandler {
```

Comment:
I suggest define a specific interfaces for `IJobHandler` & `ITaskHandler` and reduce / separate the "baseHandler" functionality to each of them